### DISCUSSION TITLE
what does "foo, err |= ..." mean

### DISCUSSION THREAD

**Discussion by tshepang** [vectordotdev/vector#11789] [Q&A] [answered]
I was surprised to see the following result in all data removed (`log(.)` results in empty object) if function fails:
```
., err |= parse_syslog(.no_exist)
```
Is this due to not understanding `|=`?

**Comment by jszwedko** [comment, marked as answer]
Hmm, no that is a bug. Thank you for reporting @tshepang . I'll open an issue for this.

**Reply by jszwedko** [reply]
Opened https://github.com/vectordotdev/vector/issues/11805

As a workaround, you can assign the result and then merge like:

```
parsed, err |= parse_syslog(.no_exist)
if err != null {
  . |= parsed
}
```
